Portal API data recoverability requirement (#37)
appuio/appuio-io-docs: docs/modules/ROOT/pages/references/quality-requirements/reliability/portal-api-data-recoverability.adoc

> = Portal API Ephemeral Data
> 
> Source::
> Portal API
> 
> Stimulus::
> Data is stored
> 
> Environment::
> Portal API
> 
> Artifact::
> Portal API and external third-party systems
> 
> Response::
> Portal API stores the data in an ephemeral storage while data persistency is granted by the third-party backend system
> 
> Rationale::
> The Portal API doesn't store any relevant data persistent.
> All relevant data is available and stored in third-party backend systems.
> The source of truth of the data is outside of the Portal API.

Reviewer: ```suggestion
= Data compatibility with external components

Source::
{product} administrator

Stimulus::
Reinstall APPUiO Global from scratch

Environment::
{zone}

Artifact::
Portal API, external third-party component

Response::
Entities that have their lifecycle outside of {product} have their availability restored

Response measure::
Following entities are recovered at latest after 48 hours since reinstallation:
+
* Invoices

Rationale::
{product} doesn't store any relevant third-party entities.
Although some form of caching is permissible, the cache can be rebuilt entirely in the given timeframe by reintegrating the third-party component.
The source of truth of the mentioned entities is outside of {product}.
```

= Data compatibility with external components

Source::
{product} administrator

Stimulus::

> Reinstall APPUiO Global from scratch

Reviewer: I think I know where the misunderstanding could come from. "Reinstall APPUiO Global from scratch" is too broad...
```suggestion
Reinstall APPUiO Global third-party component from scratch
```

Environment::
{zone}

Artifact::
Portal API, external third-party component

Response::
Entities that have their lifecycle outside of {product} have their availability restored

Response measure::

> Following entities are recovered at latest after 48 hours since reinstallation:

Author: ```suggestion
Following entities are recovered at latest after 8 hours since reinstallation:
```

I feel 48 hours is way too long, it should be possible to restore much faster.

+

> * Invoices

Author: ```suggestion
* Invoices
* List of APPUiO Zones
* Organizations
* Teams
* Users
```

I really don't like to have this list here as I'm pretty sure that this list falls behind very fast. I'd rather find a generic description of this list, like "APPUiO Cloud Core Entities".

"The {product} core entities are recovered at latest after X hours since reinstallation"

We can then have another page describing the core entities which we can amend as the architecture grows without having to change a requirement.

Reviewer: I think we may have a misunderstanding. Organizations, teams and users are stored in Keycloak and thus are not external data. To my understanding, this requirement doesn't need to cover those. Otherwise you'd be addressing disaster recovery processes, right? We haven't talked about that yet.
This requirement is about restoring data and processes where APPUiO Cloud has no data authority thereof.
